lib/galaxy/datatypes/converters/biom.xml

```diff
@@ -0,0 +1,49 @@
+<tool id="CONVERTER_biom" name="Convert Biom datasets" version="2.1.5">
+    <requirements>
+        <requirement type="package" version="2.1.5">biom-format</requirement>
+    </requirements>
+    <command><![CDATA[
+        biom convert -i '$input' -o '$output'
+        #if $__target_datatype == 'biom2'
+            --to-hdf5
+        #elif $__target_datatype == 'biom1'
+            --to-json
+        #else
+            >&2 echo "BIOM converter received unknown target datatype: $__target_datatype" &&
+            exit 1
+        #end if
+    ]]></command>
+    <configfiles>
+        <configfile name="ext_config">{"output": {
+  "name": "$input.name converted to $__target_datatype",
+  "ext": "$__target_datatype"
+}}</configfile>
+    </configfiles>
+    <inputs>
+        <param name="input" type="data" format="biom1,biom2" label="Biom file"/>
+        <!-- this hidden parameter is set automatically when the tool is called as
+             converter and set manually when called from the tool test -->
+        <param name="__target_datatype" type="hidden" label="Target data type"/>
```

Comment:
Can you make this a normal select parameter? That wait you can skip the else branch (`BIOM converter received unknown target datatype: $__target_datatype ...`) and users can set the parameter in the user interface.

Reply:
Yep. Where in the UI could users select the target datatype? In the dataset properties convert tab?

Reply:
The plan is to expose all converters, but for now https://usegalaxy.org/root?tool_id=sam_to_bam

Reply:
Thanks for the explanation. In this case I guess the biom converter should be excluded https://github.com/galaxyproject/tools-iuc/blob/master/tools/biom_format/biom_convert.xml :)